README.md

```diff
@@ -15,33 +16,111 @@ npm install equihash
 ```
 
 ## The Equihash API
-- solve(input, options, callback(err, proof))
-- verify(input, proof)
+
+`solve(seed, options, callback(err, proof))`
+
+Solve for a single solution.
+
+`seed`: buffer of bytes to use as a seed
+
+`options` (common):
+- `n`: equihash `n` parameter
+- `k`: equihash `k` parameter
+- `nonceLength`: number of bytes of nonce data to find (optional)
+
+`options` (engine specific):
+- `personal`: buffer of "personal" bytes used in some blake2 initializations
+  (optional)
+- `nonce`: initial value of nonce (engine dependent type) (optional)
+- `maxNonces`: max number of nonces to check (optional)
+
+`proof`:
+- `n`: equihash `n` parameter
+- `k`: equihash `k` parameter
+- `personal`: buffer of bytes used for "personal" bytes (engine dependent, optional)
```

Comment:
Engine specific output should be nested under the engine name as a property. Potentially also nested under "engine".

Reply:
Perhaps the engine name itself should also be included in the proof -- as an "algorithm". So "algorithm": "khot...". Then, from that, "algorithmParameters" can include any algorithm-specific parameters like "personal".